src/tests/ftest/util/apricot/apricot/test.py

```diff
@@ -132,17 +130,87 @@ def __init__(self, *args, **kwargs):
         # use 'add_cancel_ticket(<ticket>)' to add to this set.
         self._teardown_cancel = set()
         self._teardown_errors = []
+        self.basepath = None
+        self.prefix = None
+        self.ofi_prefix = None
 
     def setUp(self):
         """Set up each test case."""
+        # get paths from the build_vars generated by build
+        with open('../../.build_vars.json') as build_vars:
+            build_paths = json.load(build_vars)
+        self.basepath = os.path.normpath(os.path.join(build_paths['PREFIX'],
+                                                      '..') + os.path.sep)
+        self.prefix = build_paths['PREFIX']
+        try:
+            self.ofi_prefix = build_paths['OFI_PREFIX']
+        except KeyError:
+            self.ofi_prefix = "/usr"
+        self.skip_from_list()
         self.check_variant_skip()
         self.log.info("*** SETUP running on %s ***", str(detect()))
         super(Test, self).setUp()
 
-    def check_variant_skip(self):
+    def skip_from_list(self):
+        """Check if test is in skip list"""
+        skip_list = ""
+        try:
+            skip_list = open("/scratch/CI-skip-list").read()
+        except Exception as excpt: # pylint: disable=broad-except
+            print("Unable to read skip list: ", excpt)
+            print("Trudging on without skipping known failing tests")
+            return
```

Comment:
The goal of the _broad-except_ was to ensure that no issues (predictable or unpredictable) with skip-list processing causes a PR to fail, and instead at worst, it hits a known failing test.

In that regard, other than using `os.path.join()` is there any other reason for your code suggestion?

Reply:
The problem with a broad except is we do not know if there is an issue - which is why they are generally a bad idea.  Since it is reading a local file the `FileNotFoundError` should be the only exception raised.  We have many examples of `with open(<file>)` in our code without `try ... except` clauses that do not cause problems.

Using `with open(<file>, <mode>) as handle:` is highly suggested as it handle the file closure for you.